### DISCUSSION TITLE
Confusion with require(_randomness > 0) in Lesson 7: SmartContract Lottery

### DISCUSSION THREAD

**Discussion by fordicus** [smartcontractkit/full-blockchain-solidity-course-py#1385] [Q&A] [answered]
The method we're using from the `VRFConsumerBase` contract is overridden. This reads
```
function fulfillRandomness(bytes32 _requestId, uint256 _randomness) internal override {
/* TODO */
}
```

As far as I understood, this is a callback function, so it will run only when the ChainLink server has returned a random number we have been waiting for. The example source code in the lecture checks if the delivered number is strictly positive. This reads
```
require(_randomness > 0, "random-not-found");
```

If this callback runs only once when the random number is returned, such `require` line appears to be misleading because a random number can be zero as `uint256` type variable, if there is no internal constraint over this value from ChainLink's side. Of course, the probability for the method to return actually zero is nearly zero, but should we really use this `require` statement inside the callback? Or, does this line impose that this overridden method can actually be called several times before the random number arrives? It shouldn't, right?

**Comment by cromewar** [comment, marked as answer]
Hello @fordicus how a interesting question

>  should we really use this require statement inside the callback?

The callback could also fail (really not likely, but not imposible) may be the transaction get reverted on the blockchain, so this require is just to verify the actual random number was gotten and also (read bellow)

> does this line impose that this overridden method can actually be called several times before the random number arrives?

Yes, the fulfill randomness is an asynchronous call so you must continue with that function just once the callback was successfully made. Otherwise you will spend a lot of gas on a transaction which isn't already completed.
